### PR TITLE
fix unsafe indexing in _embed_bonds when gather_idxs is empty (Issue #558)

### PR DESCRIPTION
**Pull request by mugiwaraluffy56**
Fixes a crash in Evoformer._embed_bonds caused by unsafe indexing when
tokens_to_{polymer,ligand}_bond gather indices contain no valid rows.

Previously, the code performed:

    contact_matrix = contact_matrix.at[gather_idxs[:, 0], gather_idxs[:, 1]].set(1.0)

This fails when gather_idxs has shape (0, 2), or when all rows are padded
and masked into (0, 0). Under JAX JIT this produces an indexing error,
causing AlphaFold3 to crash on pure-protein systems or ligands whose
bond information is missing.

This patch:

• Uses gather_mask to compute a Boolean valid_rows mask  
• Selects only valid gather indices  
• Wraps the scatter_update in jax.lax.cond to safely skip empty inputs  
• Preserves original behavior for non-empty bond sets  

This resolves Issue #558 and makes bond embedding robust to missing or
incomplete bond annotations.

### CODE REVIEW

**Comment by mugiwaraluffy56** [comment]
this pr is ready , can u review @Augustin-Zidek
